resources/session03/wsgi/calculator.py

```diff
@@ -0,0 +1,66 @@
+import re
+
+
+def resolve_path(path):
+    urls = [(r'^$', welcome),
+            (r'^add/(\d+)/(\d+)$', add),
+            (r'^subtract/(\d+)/(\d+)$', subtract),
+            (r'^divide/(\d+)/(\d+)$', divide),
+            (r'^multiply/(\d+)/(\d+)$', multiply)]
+    
+    matchpath = path.lstrip('/')
+    for regexp, func in urls:
+        match = re.match(regexp, matchpath)
+        if match is None:
+            continue
+        args = match.groups([])
+        return func, args
+    # we get here if no url matches
+    raise NameError
+
+def welcome():
+    return "<h1>welcome</h1>"
+
+def multiply(v1, v2):
+    r = int(v1) * int(v2)
+    return "<h1>result from multiplication: %d</h1>" %r
+
+def add(v1, v2):
+    r = int(v1) + int(v2)
+    return "<h1>result from addition: %d</h1>" % r  
+
+def subtract(v1, v2):
+    r = int(v1) - int(v2)
+    return "<h1>result from subtraction: %d</h1>" % r
+
+def divide(v1, v2):
+    r = int(v1) / int(v2)
+    return "<h1>result from division: %d</h1>" % r
+
+def application(environ, start_response):
+    headers = [("Content-type", "text/html")]
+    try:
+        path = environ.get('PATH_INFO', None)
+        if path is None:
+            raise NameError
+        func, args = resolve_path(path)
```

Comment:
With a different approach to dispatch, you could save yourself needing to apply a function to args, like is happening here.  Feel free to play with alternate approaches.